### ISSUE TITLE
Init MovieFileClip with python file object instead of name

### ISSUE THREAD

**Issue by taddyhuo** [Zulko/moviepy#500]
I try to use MoviePy for my python flask web server. I got the video file passed in from request:
```
class MediaLibraryAPI(Resource):
    def post(self):
        print(request.files.values()[0]) # <--- <FileStorage: u'2.mp4' ('video/mp4')>
        file_storage = request.files.values()[0]
        print(file_storage.stream)  # <--- <open file '<fdopen>', mode 'wb+' at 0x7fb7fd03c270>
        video = VideoFileClip(file_storage.stream)
```
That's what I tried to do and for sure that doesn't work because the filename should be string and something like "2.mp4" with the correct path.

However, I don't want to save the video from memory to my server disk temporarily and use ffmpeg_reader to read the file from disk to memory again. 

Is there any work around for my case? 
Thanks.

**Comment by earney** [comment]
@taddyhuo   I do not think this is currently possible with the current api's.   I do think this is an interesting idea, and I've placed the enhancement label on this issue.  Maybe we will implement this sometime in the future.

**Comment by Gloin1313** [comment]
Sorry, [enhancement] is the wrong tag, [new-feature] is what you want.

**Comment by earney** [comment]
Thanks!

**Comment by travega** [comment]
Has there been any movement on this? I really need this feature as I have no disk storage to work with!

**Comment by doersino** [comment]
Bumping this issue – it would be helpful for my use case, too.

**Comment by tburrows13** [comment]
If you can represent the file stream as a numpy array, then you can create a VideoClip for that. I'd like to add a better API for this, but this will come after v2.0, as we are focusing on the necessary breaking changes first.

**Comment by questjay** [comment]
did any one solve this?

**Comment by OsaAjani** [comment]
Thank you for your contributions and for reporting issues in this repository. With the release of v2, which introduces significant changes to the codebase and API, we’ve reviewed the backlog of open PRs and issues. Due to the length of the backlog and the likelihood that many of these are either fixed or no longer applicable, we’ve made the decision to close all previous PRs and issues.

If you believe that any of these are still relevant to the current version or if you'd like to reopen a related discussion, please feel free to create a new issue or pull request, referencing the old one.

Thank you for your understanding and continued support!
